redis/src/main/java/org/hibernate/ogm/datastore/redis/RedisDialect.java

```diff
@@ -0,0 +1,546 @@
+		}
+		else {
+			return null;
+		}
+	}
+
+	private boolean isInTheQueue(EntityKey key, TupleContext tupleContext) {
+		OperationsQueue queue = tupleContext.getOperationsQueue();
+		return queue != null && queue.contains( key );
+	}
+
+	@Override
+	public Tuple createTuple(EntityKey key, TupleContext tupleContext) {
+		// TODO we don't verify that it does not yet exist assuming that this has been done before by the calling code
+		return new Tuple( new RedisTupleSnapshot( new HashMap<String, Object>() ) );
+	}
+
+	@Override
+	public void insertOrUpdateTuple(EntityKey key, Tuple tuple, TupleContext tupleContext) {
+		Map<String, Object> map = ( (RedisTupleSnapshot) tuple.getSnapshot() ).getMap();
+		MapHelpers.applyTupleOpsOnMap( tuple, map );
+		storeEntity( key, map, tupleContext.getOptionsContext() );
+	}
+
+	@Override
+	public void removeTuple(EntityKey key, TupleContext tupleContext) {
+		remove( key );
+	}
+
+	@Override
+	public boolean isStoredInEntityStructure(
+			AssociationKeyMetadata keyMetadata,
+			AssociationTypeContext associationTypeContext) {
+
+		AssociationStorageType associationStorage = getAssociationStorageType( associationTypeContext );
+
+		if ( keyMetadata.isOneToOne() || keyMetadata.getAssociationKind() == AssociationKind.EMBEDDED_COLLECTION || associationStorage == AssociationStorageType.IN_ENTITY ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	private AssociationStorageType getAssociationStorageType(AssociationTypeContext associationTypeContext) {
+		return associationTypeContext.getOptionsContext().getUnique(
+				AssociationStorageOption.class
+		);
+	}
+
+
+	private Long getTTL(OptionsContext optionsContext) {
+		return optionsContext.getUnique( TTLOption.class );
+	}
+
+	private Long getTTL(AssociationContext associationContext) {
+		return associationContext.getAssociationTypeContext().getOptionsContext().getUnique( TTLOption.class );
+	}
+
+	protected EntityStorageStrategy getEntityStorageStrategy(OptionsContext optionsContext) {
+		EntityStorageType entitiyStorage = optionsContext.getUnique(
+				EntityStorageOption.class
+		);
+
+		return getEntityStorageStrategy( entitiyStorage );
+	}
+
+	public EntityStorageStrategy getEntityStorageStrategy(EntityStorageType entitiyStorage) {
+		return entityStorageStrategies.get( entitiyStorage );
+	}
+
+	@Override
+	public Number nextValue(NextValueRequest request) {
+		byte[] key = prepareKey( request.getKey().getColumnNames(), request.getKey().getColumnValues() );
+		byte[] hget = connection.hget( IDENTIFIERS_AS_BYTES, key );
+
+		if ( hget == null || hget.length == 0 ) {
+			connection.hset( IDENTIFIERS_AS_BYTES, key, toBytes( Long.toString( request.getInitialValue() ) ) );
+			return request.getInitialValue();
+		}
+
+		return connection.hincrby( IDENTIFIERS_AS_BYTES, key, request.getIncrement() );
+	}
+
+	@Override
+	public void forEachTuple(final ModelConsumer consumer, EntityKeyMetadata... entityKeyMetadatas) {
+
```

Comment:
We usually don't do empty lines after the method name. I don't care too much, it's just a uncommon thing which makes your flow stop for a sec when reading the code.